app/views/dump.py

```diff
@@ -27,8 +27,8 @@ def dump_answers():
 def dump_submission():
     answer_store = get_answer_store(current_user)
     metadata = get_metadata(current_user)
-    schema = load_schema_from_metadata(metadata)
+    g.schema = load_schema_from_metadata(metadata)
```

Comment:
This works, but it doesn't seem ideal that `get_full_routing_path` relies on request variables that have to be manually set before calling it.  Is it non-trivial to refactor it?  Could `g.schema` perhaps be set more automatically e.g. as part of retrieving a session?

Reply:
The pattern to retrieve any other global is via a `get_x` method in `app.globals`. These access `current_user` as an argument. I think it'd be much safer to only access `g.schema` via one of these methods.

_aside_

The preferable solution would be for the function in question in `rules.py` to have an explicit dependency so `dump_submission` would be able to control directly the schema that's set.

As a rule I don't think it's a good idea for us to use globals outside of view functions - there are only a few exceptions to this which I think should be refactored.